### PR TITLE
#130: Item batcher test cases and validation logic

### PR DESCRIPTION
**Pull request by tvhees**

### CODE REVIEW

File: src/schemas/map.json [hunk 1]
```diff
@@ -86,21 +86,49 @@
...
+          "required": ["MaxItemsPerBatchPath"]
+        },
+        {
+          "required": ["MaxInputBytesPerBatch"]
+        },
+        {
+          "required": ["MaxInputBytesPerBatchPath"]
+        }
+      ],
+      "allOf": [
```
Comment (tvhees) [inline src/schemas/map.json:123]: Prevents specification of both static and path-based properties together. I'm not certain if this is the best way to do it - the error message is not particularly helpful if you violate this rule: `"Message": "/States/Map/ItemBatcher is invalid. must NOT be valid",`

File: src/schemas/map.json [hunk 2]
```diff
@@ -86,21 +86,49 @@
...
+          "maximum": 262144
+        },
+        "MaxInputBytesPerBatchPath": {
+          "$ref": "paths.json#/definitions/asl_ref_path"
+        },
+        "BatchInput": {
+          "$ref": "paths.json#/definitions/asl_payload_template"
+        }
+      },
+      "anyOf": [
```
Comment (tvhees) [inline src/schemas/map.json:109]: Ensures at least one batch constraint is specified.

**Review by ChristopheBougere — CHANGES_REQUESTED** [review]
Thanks !
I just rebased after the merge of #129 (and I broke commit link, sorry about that).
The combination of `allOf`, `oneOf`, `not` was starting to get messy with all the checks in the project, so I think it would be great to use the new check system than here in the `stateChecks` function:
https://github.com/ChristopheBougere/asl-validator/blob/main/src/validator.ts#L40
(keep the new `properties` field in the schema, but move the `oneOf`/`allOf` logic in the TS file)

**Comment by ChristopheBougere** [comment]
> @ChristopheBougere for clarity do you mean a new entry such as
> 
> `errors.push(...itemBatcherErrors(definition, options));`
> 
> I will develop a solution like that.
> 
> I initially thought you meant a new entry in the `stateChecks` array but I don't see how the existing checkers can operate on subproperties of states.

Well, I was also thinking about new entries in the `stateChecks` array but I didn't think about the nested issue.

Maybe you could update the [AtMostOne](https://github.com/ChristopheBougere/asl-validator/blob/main/src/checks/state-checks.ts#L63) and [ExactlyOne](https://github.com/ChristopheBougere/asl-validator/blob/main/src/checks/state-checks.ts#L81) functions to take an optional property JSON path as argument, so it would work for both root and nested fields.
Or maybe it's better to write new state checkers if it complexify the code too much.

**Comment by ChristopheBougere** [comment]
Thanks for the update @tvhees 
You've been pushing very similar update allowing to specify a json path in AtMostOne in a short time frame with @massfords :
https://github.com/ChristopheBougere/asl-validator/pull/133/files#diff-845bbdbb49ce1001585d99734e8c2592d7712dc0f26021d9b403773ef7150716R64

Would you mind rebase your branch, undo the change in `state-checks.ts` and use the new `path` property of `AtMostOne`?
I think it should do the job!

**Pull request closed without merging**